src/diff.ts

```diff
@@ -379,31 +384,32 @@ export function visualDomDiff(
     }
 
     // Move deletes before inserts.
-    for (removedNode of removedNodes) {
-        const parentNode = removedNode.parentNode as Node
-        let previousSibling = removedNode.previousSibling
+    removedNodes.forEach(node => {
+        const parentNode = node.parentNode as Node
+        let previousSibling = node.previousSibling
 
         while (previousSibling && addedNodes.has(previousSibling)) {
-            parentNode.insertBefore(removedNode, previousSibling)
-            previousSibling = removedNode.previousSibling
+            parentNode.insertBefore(node, previousSibling)
+            previousSibling = node.previousSibling
         }
-    }
+    })
 
     // Ensure a user friendly result for tables.
-    for (const { newTable, oldTable, outputTable } of equalTables) {
+    equalTables.forEach(equalTable => {
+        const { newTable, oldTable, outputTable } = equalTable
         // Handle tables which can't be diffed nicely.
         if (
             !isTableValid(oldTable, true) ||
             !isTableValid(newTable, true) ||
             !isTableValid(outputTable, false)
         ) {
             // Remove all values which were previously recorded for outputTable.
-            for (const node of new DomIterator(outputTable)) {
+            new DomIterator(outputTable).toList().forEach(node => {
```

Comment:
I wonder if implementing `forEach`, `some` and `reduce` directly on `DomIterator` would improve performance... it would certainly avoid allocating temporary arrays.

Reply:
I'm not sure how perf gain it will get either. It probably doesn' affect the time to run the scripts but the memory allocation? But clearly bundle size will increase if we implement these functions isn't it?